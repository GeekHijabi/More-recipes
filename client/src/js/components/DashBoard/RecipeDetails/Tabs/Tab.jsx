import React from 'react';

const Tab = props => (
  <a
    className={
      props.isActive ?
        'nav-item nav-link active' :
        'nav-item nav-link'}
    id="ingredients-tab"
    data-toggle="tab"
    role="tab"
    aria-controls="ingredients"
    aria-selected="false"
    onClick={props.handleClick}
    href={`#${props.tabOption}`}
  >
    {props.tabOption.name}
  </a>
);

export default Tab;
