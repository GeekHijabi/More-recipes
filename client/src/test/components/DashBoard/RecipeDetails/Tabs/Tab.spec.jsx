import React from 'react';
import { shallow } from 'enzyme';
import Tab
  from '../../../../../js/components/DashBoard/RecipeDetails/Tabs/Tab';

const props = {
  handleClick: jest.fn(() => Promise.resolve({ ok: true })),
  isActive: false,
  tabOption: {
    id: 3, name: 'Reviews', isActive: false
  }
};


describe('Tab snapshot', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<Tab {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});

