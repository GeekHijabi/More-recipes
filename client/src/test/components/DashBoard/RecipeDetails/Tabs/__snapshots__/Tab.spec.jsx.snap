// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab snapshot should render correctly 1`] = `
<a
  aria-controls="tab"
  aria-selected="false"
  className="nav-item nav-link"
  data-toggle="tab"
  href="#[object Object]"
  id="tab"
  onClick={[MockFunction]}
  role="tab"
>
  Reviews
</a>
`;
